File: lib/domain/auth/ports/auth_test_mock_login_output_port.rb
# frozen_string_literal: true

module Domain
  module Auth
    module Ports
      module AuthTestMockLoginOutputPort
        def on_environment_forbidden
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_missing_mock
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_create_failed(error_messages:)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_success_process_saved_plan(session_id:, expires_at:)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_success_return_to(url:, session_id:, expires_at:, user_name:)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end

        def on_success_root(session_id:, expires_at:, user_name:)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/auth/ports/auth_test_mock_login_output_port.rs
use time::OffsetDateTime;

/// Ruby: `Domain::Auth::Ports::AuthTestMockLoginOutputPort`
pub trait AuthTestMockLoginOutputPort {
    fn on_environment_forbidden(&mut self);
    fn on_missing_mock(&mut self);
    fn on_create_failed(&mut self, error_messages: Vec<String>);
    fn on_success_process_saved_plan(&mut self, session_id: &str, expires_at: OffsetDateTime);
    fn on_success_return_to(
        &mut self,
        url: &str,
        session_id: &str,
        expires_at: OffsetDateTime,
        user_name: &str,
    );
    fn on_success_root(
        &mut self,
        session_id: &str,
        expires_at: OffsetDateTime,
        user_name: &str,
    );
}
